.github/workflows/codespell.yaml

```
@@ -0,0 +1,24 @@
+name: codespell
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+jobs:
+  codespell:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

I do not think we need to fetch all history for all branches and tags

```suggestion
        with:
          fetch-depth: 0
```